pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -257,21 +252,22 @@ func (rq *ResourceQuotaController) syncResourceQuota(quota api.ResourceQuota) (e
 		usage.Status.Used[k] = *v.Copy()
 	}
 
-	set := map[api.ResourceName]bool{}
-	for k := range usage.Status.Hard {
-		set[k] = true
+	// recalculate usage
+	usageOptions := UsageOptions{
+		Namespace: quota.Namespace,
+		Resources: resourceSet,
 	}
-
-	pods := &api.PodList{}
-	if set[api.ResourcePods] || set[api.ResourceMemory] || set[api.ResourceCPU] {
-		pods, err = rq.kubeClient.Pods(usage.Namespace).List(api.ListOptions{})
+	latestUsage := api.ResourceList{}
+	for _, usageFunc := range rq.usageRegistry.UsageFuncs() {
+		usage, err := usageFunc(usageOptions)
 		if err != nil {
 			return err
 		}
+		for k, v := range usage.Used {
+			latestUsage[k] = v
```

Comment:
What about a case that 2 usage funcs update the same resource? I think that shouldn't happen and should be logged as an error. Also shouldn't we pick the maximum value?